Content.Server/Nuke/NukeCodePaperSystem.cs

```diff
@@ -86,5 +68,35 @@ public bool SendNukeCodes(EntityUid station)
 
             return wasSent;
         }
+
+        private bool TryGetRelativeNukeCode(
+            EntityUid uid,
+            [NotNullWhen(true)] out string? nukeCode,
+            EntityUid? station = null,
+            TransformComponent? transform = null)
+        {
+            nukeCode = null;
+            if (!Resolve(uid, ref transform))
+            {
+                return false;
+            }
+            
+            var owningStation = station ?? _station.GetOwningStation(uid);
+            
+            // Find the first nuke that matches the passed location.
+            foreach (var nuke in EntityQuery<NukeComponent>())
+            {
+                if (owningStation == null && nuke.OriginMapGrid != (transform.MapID, transform.GridUid)
+                    || nuke.OriginStation != owningStation)
+                {
+                    continue;
+                }
+
+                nukeCode = $"{MetaData(nuke.Owner).EntityName} - {nuke.Code}";
```

Comment:
Now you're missing the text from the original nuke code message. You sould add that back in:
```csharp
$@"
[color=red]TOP SECRET![/color]
Nuclear device activation code: {MetaData(nuke.Owner).EntityName} - {nuke.Code}
"
````